### PR TITLE
[mxfp8 moe training] Add mxfp8 to FSDP tests

### PR DESCRIPTION
**Pull request by liangel-02**
addressing #2833 

updating test to include mxfp8: `torchrun --nproc_per_node=${NUM_GPUS} -m pytest test_fsdp.py `, all tests pass

### CODE REVIEW

**Comment by danielvegamyhre** [comment]
@liangel-02 we need to use this torchtitan API:

```python
set_token_group_alignment_size_m(16) # fp8
```

or

```python
set_token_group_alignment_size_m(32) # mxfp8
```


This is because TMA (some background [here](https://pytorch.org/blog/hopper-tma-unit/)) requires the slowest moving dim (stride of 1) to be 128 bit (16 byte) aligned. 

In the backward pass, when grad_weight = grad_output_t @ input, the “M” dimension (flattened token groups) become this “stride 1” dim. Therefore, each token group must be 16 byte aligned for this grouped gemm. 

- For bf16, this means groups divisible by 8 elements (16 bytes / 2 bytes per elem).
- For fp8 groups divisible by 16 elements  (16 bytes / 1 byte per elem).  
- For mxfp8, I enforce they must be divisible by `block_size` (32) not because of TMA, but to ensure when we quantize along that dim, we have (1) no scaling groups crossing logically distinct tensor boundaries, and (2) no “partial” scaling group at the the end (e..g, 1x17 chunk because the tensor ends).

This might all be a bit confusing without background knowledge on GPU architecture and MoE models, but don't worry! Plenty of time to learn.

For now, just implement this setting in some clean/interpretable way in the test code and it should (hopefully) work.

File: test/prototype/moe_training/test_fsdp.py [hunk 1]
```diff
@@ -83,7 +95,8 @@ def moe_module_filter_fn(mod: nn.Module, cur_fqn: str) -> bool:
         return False
 
     # quantize test model
-    config = MoETrainingConfig()
+    config = MoETrainingConfig(recipe)
+    # config = MoETrainingConfig()
```
Comment (danielvegamyhre) [inline test/prototype/moe_training/test_fsdp.py]: remove commented code before landing

**Review by danielvegamyhre — APPROVED** [review]
lgtm! 1 minor comment before landing, thanks
